--- install.py
+++ install.py
@@ -761,9 +761,6 @@ def main(argv):
                     (v[0], v[1], v[2])
                 print "    with foreign characters. Please see 'README-PyXML.txt' for"
                 print "    details."
-                res = raw_input("    Shall pyXML be patched now (y/n)? ")
-                if res in ["Y", "y"]:
-                    patchPyXML()
     except ImportError:
         pass
     
