--- designer-Qt3/designer.pro
+++ designer-Qt3/designer.pro
@@ -3,10 +3,13 @@
 
 TEMPLATE = lib
 TARGET = qscintillaplugin
-DESTDIR = $(QTDIR)/plugins/designer
 
 CONFIG += qt warn_on release plugin
 
 SOURCES += qscintillaplugin.cpp
 
-LIBS += -lqscintilla2
+target.path = $(QTDIR)/plugins/designer
+INSTALLS += target
+
+INCLUDEPATH = ../Qt3
+LIBS += -L../Qt3 -lqscintilla2
--- designer-Qt4/designer.pro
+++ designer-Qt4/designer.pro
@@ -12,4 +12,5 @@
 target.path = $$[QT_INSTALL_PLUGINS]/designer
 INSTALLS += target
 
-LIBS += -lqscintilla2
+INCLUDEPATH = ../Qt4
+LIBS += -L../Qt4 -lqscintilla2
