--- setup.py
+++ setup.py
@@ -281,6 +281,8 @@ class pil_build_ext(build_ext):
         if struct.unpack("h", "\0\1")[0] == 1:
             defs.append(("WORDS_BIGENDIAN", None))
 
+        libs.append("m")
+
         exts = [(Extension(
             "_imaging", files, libraries=libs, define_macros=defs
             ))]
@@ -332,7 +334,7 @@ class pil_build_ext(build_ext):
                 ))
 
         if os.path.isfile("_imagingmath.c"):
-            exts.append(Extension("_imagingmath", ["_imagingmath.c"]))
+            exts.append(Extension("_imagingmath", ["_imagingmath.c"], libraries=["m"]))
 
         self.extensions[:] = exts
 
